packages/webapp/src/members/components/withdraw-funds/withdraw-modal-step-form.tsx

```diff
@@ -0,0 +1,150 @@
+
+    const amountInputPreventChangeOnScroll = (
+        e: React.WheelEvent<HTMLInputElement>
+    ) => (e.target as HTMLInputElement).blur();
+
+    return (
+        <div className="space-y-4">
+            <Heading>Withdraw funds</Heading>
+            <Text>
+                Withdraw my funds from the Eden contract to a public{" "}
+                {tokenConfig.symbol} account.
+            </Text>
+            <Text>
+                Available:{" "}
+                <span className="font-medium">
+                    {assetToLocaleString(
+                        availableFunds,
+                        availableFunds.precision
+                    )}
+                </span>
+            </Text>
+            <form onSubmit={onPreview} className="space-y-3">
+                <Form.LabeledSet
+                    label={`${tokenConfig.symbol} account name (12 characters)`}
+                    htmlFor="to"
+                >
+                    <Form.Input
+                        id="to"
+                        type="text"
+                        required
+                        value={fields.to}
+                        onChange={onChangeFields}
+                        maxLength={12}
+                        pattern="^[a-z,1-5.]{1,12}$"
+                        onInvalid={validateAccountField}
+                        onInput={clearErrorMessages}
+                    />
+                </Form.LabeledSet>
+                <Form.LabeledSet label="Amount to withdraw" htmlFor="amount">
+                    <div className="flex space-x-2">
+                        <div className="relative flex-1">
+                            <Form.Input
+                                id="amount"
+                                type="number"
+                                inputMode="decimal"
+                                min={1 / Math.pow(10, tokenConfig.precision)}
+                                max={maxWithdrawal}
+                                step="any"
+                                required
+                                value={fields.amount}
+                                onChange={onChangeFields}
+                                maxLength={12}
+                                onWheel={amountInputPreventChangeOnScroll}
+                                onInvalid={validateAmountField}
+                                onInput={clearErrorMessages}
+                            />
+                            <div className="absolute top-3 right-2">
+                                <p className="text-sm text-gray-400">
+                                    {availableFunds?.symbol}
+                                </p>
+                            </div>
+                        </div>
+                        <Button type="neutral" onClick={setMaxAmount}>
+                            Max
+                        </Button>
+                    </div>
+                </Form.LabeledSet>
+                {isThirdPartyWithdrawal && (
```

Comment:
Question: Why not allow memo to transfers being sent to itself? Might be helpful for the user history and helps on community visibility

Reply:
I agree, but the `eosio.token:transfer` action is only called when you transfer funds to a third-party account, and for the time being, no matter what amount you specify for withdrawal/transfer, we sweep all available distribution funds to your personal Eden account (not a partial amount)...so the `fundtransfer` action (which does contain a memo field) might be for a much higher token amount than the `withdraw` and `transfer`. So a memo on the withdrawal itself could end up inaccurate.

We have enhanced fund management on the roadmap, and these are the types of things we'll want to tackle then.

Reply:
oh, I missed that the memo field was only for the `eosio::transfer`! now it makes sense!